tools/make-v8.sh

```diff
@@ -1,38 +1,47 @@
 #!/bin/bash
 
-
-git_origin=$(git config --get remote.origin.url | sed 's/.\+[\/:]\([^\/]\+\/[^\/]\+\)$/\1/')
-git_branch=$(git rev-parse --abbrev-ref HEAD)
-v8ver=${1:-v8} #default v8
-svn_prefix=https://github.com
-svn_path="$svn_prefix/$git_origin/branches/$git_branch/deps/$v8ver"
-#svn_path="$git_origin/branches/$git_branch/deps/$v8ver"
-gclient_string="solutions = [{'name': 'v8', 'url': '$svn_path', 'managed': False}]"
+# Get V8 branch from v8/include/v8-version.h
+MAJOR=$(grep V8_MAJOR_VERSION deps/v8/include/v8-version.h | cut -d ' ' -f 3)
+MINOR=$(grep V8_MINOR_VERSION deps/v8/include/v8-version.h | cut -d ' ' -f 3)
+BRANCH=$MAJOR.$MINOR
 
 # clean up if someone presses ctrl-c
 trap cleanup INT
 
 function cleanup() {
   trap - INT
-
   rm .gclient || true
   rm .gclient_entries || true
   rm -rf _bad_scm/ || true
-
-  #if v8ver isn't v8, move the v8 folders
-  #back to what they were
-  if [ "$v8ver" != "v8" ]; then
-    mv v8 $v8ver
-    mv .v8old v8
-  fi
+  echo "git cleanup"
+  git reset --hard HEAD
+  git clean -e .v8old -ffdq
+  # Copy local files
+  rsync -a .v8old/ v8/
+  rm -rf .v8old
   exit 0
 }
 
 cd deps
-echo $gclient_string > .gclient
-if [ "$v8ver" != "v8" ]; then
-  mv v8 .v8old
-  mv $v8ver v8
+# Preserve local changes
+mv v8 .v8old
+
+echo "Fetching v8 from chromium.googlesource.com"
+fetch v8
+if [ "$?" -ne 0 ]; then
+  echo "V8 fetch failed"
+  exit 1
 fi
+echo "V8 fetched"
+
+cd v8
+
+echo "Checking out branch:$BRANCH"
+git fetch
```

Comment:
If I am not wrong, `fetch v8` will fetch everything. So this may not be necessary.